codec/src/main/java/io/netty/handler/codec/compression/SnappyFrameDecoder.java

```diff
@@ -98,21 +97,24 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
 
             switch (chunkType) {
                 case STREAM_IDENTIFIER:
-                    if (chunkLength != SNAPPY.length) {
+                    if (chunkLength != SNAPPY_IDENTIFIER_LEN) {
                         throw new DecompressionException("Unexpected length of stream identifier: " + chunkLength);
                     }
 
-                    if (inSize < 4 + SNAPPY.length) {
+                    if (inSize < 4 + SNAPPY_IDENTIFIER_LEN) {
                         break;
                     }
 
-                    byte[] identifier = new byte[chunkLength];
-                    in.skipBytes(4).readBytes(identifier);
+                    in.skipBytes(4);
+                    int offset = in.readerIndex();
+                    in.skipBytes(SNAPPY_IDENTIFIER_LEN);
```

Comment:
I think it not really matters and is consistent with how it behaved beforr

> Am 24.07.2016 um 15:49 schrieb Scott Mitchell <notifications@github.com>:
> 
> In codec/src/main/java/io/netty/handler/codec/compression/SnappyFrameDecoder.java:
> 
> >                          break;
> >                      }
> >  
> > -                    byte[] identifier = new byte[chunkLength];
> > -                    in.skipBytes(4).readBytes(identifier);
> > +                    in.skipBytes(4);
> > +                    int offset = in.readerIndex();
> > +                    in.skipBytes(SNAPPY_IDENTIFIER_LEN);
> should we only skip the bytes if the check passes?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>